cares_reinforcement_learning/memory/memory_buffer.py

```diff
@@ -40,6 +40,29 @@ def sample(self, batch_size):
         transposed_batch = zip(*experience_batch)
         return transposed_batch
 
+    def sample_next(self, batch_size):
+        """
+        For training MBRL to predict rewards. The right next transition is
+        sampled as well. WHEN THE BUFFER IS NOT SHUFFLED.
+
+        (State, action, reward, next_state, next_action, next_reard)
```

Comment:
how is this different to the normal buffer exactly?

Reply:
The buffer is the same. Just this function will retrieve the next timestep's action and reward for training a world model that also predict reward.